components/gitpod-db/src/typeorm/migration/1630333424897-ReplaceStringBooleansWithBooleans.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import {MigrationInterface, QueryRunner} from "typeorm";
+import { columnExists } from "./helper/helper";
+
+export class ReplaceStringBooleansWithBooleans1630333424897 implements MigrationInterface {
+
+    public async up(queryRunner: QueryRunner): Promise<any> {
+        if (await columnExists(queryRunner, "d_b_user", "additionalData")) {
+            await queryRunner.query("update d_b_user set additionalData = json_set(additionalData, '$.emailNotificationSettings.allowsChangelogMail', true) where json_extract(additionalData, '$.emailNotificationSettings.allowsChangelogMail') = 'true'");
+            await queryRunner.query("update d_b_user set additionalData = json_set(additionalData, '$.emailNotificationSettings.allowsChangelogMail', false) where json_extract(additionalData, '$.emailNotificationSettings.allowsChangelogMail') = 'false'")
```

Comment:
I'm not very familiar with json functions in mysql, thus had to look it up first. ;-)

wdyt of using `JSON_UNQUOTE` and generalize, which would shave off the second query. I mean something like this:

```
son_set(additionalData, '$.emailNotificationSettings.allowsChangelogMail', JSON_UNQUOTE(json_extract(additionalData, '$.emailNotificationSettings.allowsChangelogMail')))
```

Reply:
Nice find with the JSON_UNQUOTE! Let's see what the DB expert @jakobhero says.

Reply:
i like the idea of the generalizing, but `JSON_QUOTE()` returns a string and the purpose of the migration is to use booleans as the consistent data type across all email notification settings. having a general solution that would use booleans could be achieved 
by checking for equality with `'true'` (cp. attached screenshot):
```
json_set(additionalData, '$.emailNotificationSettings.allowsChangelogMail', JSON_UNQUOTE(json_extract(additionalData, '$.emailNotificationSettings.allowsChangelogMail'))='true')
```
![Screenshot 2021-09-13 at 19 57 24](https://user-images.githubusercontent.com/59581255/133134259-73ccc22d-7397-4488-908a-32d53dc73bda.png)